src/prefect/flows.py

```diff
@@ -158,6 +159,26 @@ def __init__(
             List[Callable[[schemas.core.Flow, FlowRun, State], None]]
         ] = None,
     ):
+        # Validate if hook passed is list and contains callables
+        hook_types = [on_completion, on_failure, on_cancellation, on_crashed]
+        hook_names = ["on_completion", "on_failure", "on_cancellation", "on_crashed"]
+        for hook, hook_name in zip(hook_types, hook_names):
+            if hook is not None:
+                try:
+                    hook = list(hook)
+                except TypeError:
+                    raise TypeError(
+                        f"Expected iterable for '{hook_name}'; got"
+                        f" {type(hook).__name__} instead."
+                    )
+
+                for hook_item in hook:
```

Comment:
`for hook in hooks`: 
might be more intuitive (since right now `hook` is a list)

Reply:
Sure, `hook` is list, so I think I would have to rename it to `hooks` and then I can make it `for hook in hooks`